src/Microsoft.Health.Fhir.Shared.Api/Modules/FhirModule.cs

```diff
@@ -79,9 +74,47 @@ ResourceElement SetMetadata(Resource resource, string versionId, DateTimeOffset
                     {
                         FhirResourceFormat.Json, (str, version, lastModified) =>
                         {
-                            var resource = jsonParser.Parse<Resource>(str);
-
-                            return SetMetadata(resource, version, lastModified);
+                             Resource resource = null;
+
+                            Parse:
+                             try
+                            {
+                                resource = jsonParser.Parse<Resource>(str);
+                            }
+                            catch (StructuralTypeException ex)
+                            {
+                                var match = MessageChecker.Match(ex.Message);
+                                if (match.Success && match.Groups.Count == 4 && match.Groups[2].Value == "date")
+                                {
+                                    var valueToReplace = match.Groups[1].Value;
+                                    var location = match.Groups[3].Value;
+                                    var replace = valueToReplace.Substring(0, 10);
+                                    var root = FhirJsonNode.Parse(str, "Resource");
+                                    var currentNode = root;
+                                    while (currentNode != null)
+                                    {
+                                        foreach (var child in currentNode.Children())
+                                        {
+                                            if (location.StartsWith(child.Location, StringComparison.OrdinalIgnoreCase))
+                                            {
+                                                currentNode = child;
+                                                break;
+                                            }
+                                        }
+
+                                        if (currentNode.Location == location)
+                                        {
+                                            break;
+                                        }
+                                    }
+
+                                    (currentNode as FhirJsonNode).JsonValue.Value = replace;
+                                    str = root.ToJson();
```

Comment:
From here could we use `root.ToTypedElement(_modelInfoProvider.StructureDefinitionSummaryProvider).ToPoco<Resource>()`?

Reply:
I don't know what it does.
What it achive?